docs/index.md

```diff
@@ -88,34 +68,102 @@ Currently, Rust is the best-supported language for writing SpacetimeDB modules.
+
+### Module
+A SpacetimeDB **module** is an application that runs on a [host](#host).
+
+A module exports [tables](#table), which store data, and [reducers](#reducer), which allow [clients](#client) to make requests.
+
+Technically, a SpacetimeDB module is a [WebAssembly module](https://developer.mozilla.org/en-US/docs/WebAssembly) that imports a specific low-level [WebAssembly API](/webassembly-api) and exports a small number of special functions. However, the SpacetimeDB [server-side libraries](#server-side-libraries) hide these low-level details. As a developer, writing a module is mostly like writing any other C# or Rust application, except for the fact that a [special CLI tool](/install) is used to build and deploy the application.
+
+### Table
+A SpacetimeDB **table** is a database table. Tables are declared in a module's native language. For instance, in Rust, a table is declared like so:
+
+```rust
+#[spacetimedb::table(name = person, public)]
+pub struct Player {
+   id: u64,
+   name: String,
+   age: u32,
+   user: Identity,
+}
+```
+
+The contents of a table can be read by [reducers](#reducer).
+Tables marked `public` can also be read by [clients](#client).
+
+### Reducer
+A **reducer** is a function exported by a [module](#module).
+Connected [clients](#client-side-sdks) can call reducers to interact with the module.
+This is a form of [remote procedure call](https://en.wikipedia.org/wiki/Remote_procedure_call).
+Reducers can be invoked across languages. For example, a Rust [module](#module) can export a reducer like so:
+
+```rust
+#[spacetimedb::reducer]
+pub fn set_player_name(ctx: &spacetimedb::ReducerContext, id: u64, name: String) -> Result<(), String> {
+   // ...
+}
+```
+
+And a C# [client](#client) can call that reducer:
+
+```cs
+void Main() {
+   // ...setup code, then...
+   Reducer.SetPlayerName(57, "Marceline");
+}
+```
+
+These look mostly like regular function calls, but under the hood, they are cross-language RPC calls.
+
+The `ReducerContext` passed into a reducer includes information about the caller's [identity](#identity) and [address](#address).
+It also allows accessing the database and scheduling future operations.
+
+### Client
+A **client** is an application that connects to a [module](#module). Clients are written using the [client-side SDKs](#client-side-sdks).
+
+A client logs in using an [identity](#identity) and receives an [address](#address) to identify the connection.
+
+### Identity
+<!-- TODO: The following will need to be updated when we implement Tyler's Identity and Auth proposals. -->
+
+A SpacetimeDB `Identity` allows someone to log in to a SpacetimeDB module.
+
+You can configure how your module issues Identities and which Identities it trusts. A user's `Identity` is attached to every [reducer call](#reducer), and you can use this to decide what they are allowed to do.
```

Comment:
Modules don't issue identities. I think we should get someone who's more familiar with the new identity stuff, likely @jsdt , to update this section.

Reply:
Yes, this isn't written to be up-to-date with the identity proposal. I'm not sure about the implementation status of that.